recipes/theora/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+import os
+import shutil
+import stat
+from conans import ConanFile, MSBuild, AutoToolsBuildEnvironment, tools
+
+
+class TheoraConan(ConanFile):
+    name = "theora"
+    description = "Theora is a free and open video compression format from the Xiph.org Foundation"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/xiph/theora"
+    topics = ("conan", "theora", "video", "video-compressor", "video-format")
+    license = "BSD-3-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    requires = (
+        "ogg/1.3.4",
+        "vorbis/1.3.7"
+    )
+    _source_subfolder = "source_subfolder"
+    _autotools = None
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
```

Comment:
```suggestion
    def configure(self):
        if self.options.shared:
            del self.options.fPIC
```

Reply:
is it needed? hook doesn't complain about it

Reply:
Yes. 90% of CCI recipes are using it. It does not make sense offering fPIC if we only build shared with fPIC. It offers an "invalid package id". Indeed we need to improve Hooks, we only check it for Windows: https://github.com/conan-io/hooks/blob/master/hooks/conan-center.py#L715

Reply:
yes, if it is something what is absolutely required, then it should be enforced in the hook. please create an issue if you think it's needed check.
otherwise, I'd say all the recipes in bincrafters were not using it and worked just fine with no complains for years (at least I don't remember any reports about `invalid package id` errors).

Reply:
> yes, if it is something what is absolutely required, then it should be enforced in the hook. please create an issue if you think it's needed check.

Yes, I'll do.

> otherwise, I'd say all the recipes in bincrafters were not using it and worked just fine with no complains for years (at least I don't remember any reports about invalid package id errors).

Not all CCI contributors are part of Bincrafters.

Reply:
yes, I mean recipe exists from 2018 and it worked well without this fix. other recipes as well AFAIK.

Reply:
They will work. It works more like a protect for an invalid package configuration, not the exception, but the package itself. Done, the issue/PR/doc are here: https://github.com/conan-io/hooks/issues/285